cl/people_db/filters.py

```diff
@@ -249,18 +251,32 @@ class Meta:
         }
 
 
-class PartyFilter(NoEmptyFilterSet):
+class PartyFilter(NoEmptyFilterSet, FilterManyToManyMixin):
     docket = filters.RelatedFilter(
         "cl.search.filters.DocketFilter",
         field_name="dockets",
         queryset=Docket.objects.all(),
+        distinct=True,
     )
     attorney = filters.RelatedFilter(
         "cl.people_db.filters.AttorneyFilter",
         field_name="attorneys",
         queryset=Attorney.objects.all(),
+        distinct=True,
+    )
+    filter_nested_results = filters.BooleanFilter(
+        field_name="roles", method="filter_roles"
     )
 
+    # Attributes for the mixin
+    # **Important:** Keep this mapping up-to-date with any changes to
+    # RelatedFilters or custom labels in this class to avoid unexpected
+    # behavior.
+    join_table_cleanup_mapping = {
+        "attorney__docket": "docket",
```

Comment:
Would it make sense to also add support for the `role` field here?
`"attorney__role": "role"`

Reply:
The specified lookup is not valid. The `PartyFilter` class relies on a related filter for attorney lookups, limiting valid attributes to those defined in the AttorneyFilter class. In our specific case, these attributes are restricted to `id`, `date_created`, `date_modified`, `name`, `docket`, and `parties_represented`.